### PR TITLE
Improve light off error log

### PR DESCRIPTION
**Pull request by isramos**
- Improve transport error message (when the device goes offline for n times)
New message: 
`[11/18/2020, 12:58:28 AM] [homebridge-magichome-dynamic-platform] Unable to connect to 192.168.1.149 5577 (code: EHOSTDOWN)`
- Fix typos (Simultanious)
- added interfaces for device and lightTypes
- moved the long list of lightTypes to a separate file
- Tested namesWithMacAddress Flag - DONE

### CODE REVIEW

File: src/platform.ts [hunk 1]
```diff
@@ -261,26 +134,25 @@ export class HomebridgeMagichomeDynamicPlatform implements DynamicPlatformPlugin
...
-              device.uniqueId);
+              deviceBroadcast.uniqueId);
             continue;
           }
        
-          const accessory = new this.api.platformAccessory(updatedDevice.lightParameters.convenientName, uuid);
-
-          //create a new transport object so we have access to devices state
-          //this is neccessary to determine the lightVersion
-          const transport = new Transport(device.ipAddress, this.config);
```
Comment (isramos) [inline src/platform.ts:275]: defined, but never used

File: .homebridge-dev/config.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
       "deviceManagement": {
         "blacklistOrWhitelist": "blacklist",
         "blacklistedUniqueIDs": [""]
-      }
+      },
+      "uniqueNames": true
```
Comment (Zacknetic) [inline .homebridge-dev/config.json:35]: Change the name to something representing mac address. i.e. MACaddressNames, additionally place it under a category or create a new category.

**Review by Zacknetic — CHANGES_REQUESTED** [review]
Yes, this is exactly what needs to be done. Lets discuss further tomorrow.

File: src/magichome-interface/Transport.ts [hunk 3]
```diff
@@ -96,9 +101,12 @@ export class Transport {
   }
 
   async send(buffer: any, useChecksum = true, _timeout = 2000) {
+    this.log.warn(`[IGOR] send (adding to queue) data to ${this.host}`);
```
Comment (Zacknetic) [inline src/magichome-interface/Transport.ts]: Why is your name in the logs? If you'd like to comment your name next to the code block that's fine. Please remove any personally identifiable information from logs.

File: .homebridge-dev/config.json (hunk 2 above)
Comment (isramos) [reply .homebridge-dev/config.json]: done. Changed to namesWithMacAddress, and added into a category.

File: src/magichome-interface/Transport.ts (hunk 3 above)
Comment (isramos) [reply src/magichome-interface/Transport.ts]: I added my name, so it would be easy to spot for removal. It's been removed :)

**Comment by isramos** [comment]
@Zacknetic - your comments have been addressed. Ready for your review.

**Pull request merged**